### PR TITLE
tools: report_bench.sh to all benchmarks required for the report

### PR DESCRIPTION
**Pull request by janekmi**
Since `rpma_fio_bench.sh` does a lot more than it is specified for the upcoming report format I have created a separate script doing just that - runs all benchmarks required to prepare the report. All comments are very welcome.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/rpma/781)

### CODE REVIEW

**Review by grom72 — COMMENTED** [review]
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/781)* status: all files reviewed, 1 unresolved discussion (waiting on @janekmi)

___
*[tools/.gitignore, line 4 at r1](https://reviewable.io/reviews/pmem/rpma/781#-MS1GQgCBlK8mt31Bwa_:-MS1GQgD0pJCGCjykD_A:b6jcgla) ([raw file](https://github.com/pmem/rpma/blob/c978f2eac5f7b08d281725700f8ec075cce32c4c/tools/.gitignore#L4)):*
> ```
> !report_bench.sh
> ```
what if we use an alternative name that does not need this exception i.e. rpma_fio_bench_report.sh?

**Review by grom72 — COMMENTED** [review]
:goat: 

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/781#-:-MS1Gsdh0YWg-rndqD8t:bdycpsm)* status: all files reviewed, 1 unresolved discussion (waiting on @janekmi)

**Review by grom72 — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/781#-:-MS1K5n64gR0hK9BOmJ-:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @janekmi)

**Review by osalyk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/781)* status: all files reviewed, 3 unresolved discussions (waiting on @janekmi)

___
*[tools/report_bench.sh, line 17 at r1](https://reviewable.io/reviews/pmem/rpma/781#-MS1eQ18E_7Lq0VOVYfl:-MS1eQ18E_7Lq0VOVYfm:b-ejz6lf) ([raw file](https://github.com/pmem/rpma/blob/c978f2eac5f7b08d281725700f8ec075cce32c4c/tools/report_bench.sh#L17)):*
> ```Shell
> 	echo "Usage: $0 <server_ip>"
>     echo
> 	echo "export REMOTE_JOB_MEM_PATH=/path/to/mem (required)"
> ```

What about REMOTE_DIRECT_WRITE_TO_PMEM?
___
*[tools/report_bench.sh, line 31 at r1](https://reviewable.io/reviews/pmem/rpma/781#-MS1defs6a1MdPAd7loh:-MS1defs6a1MdPAd7loi:bb2anwo) ([raw file](https://github.com/pmem/rpma/blob/c978f2eac5f7b08d281725700f8ec075cce32c4c/tools/report_bench.sh#L31)):*
> ```Shell
> export SERVER_IP=$1
> export PMEM=$REMOTE_JOB_MEM_PATH
> export DRAM="malloc"
> ```

Isn't this variable redundant?

**Review by janekmi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/781)* status: all files reviewed, 3 unresolved discussions (waiting on @grom72 and @osalyk)

___
*[tools/.gitignore, line 4 at r1](https://reviewable.io/reviews/pmem/rpma/781#-MS1GQgCBlK8mt31Bwa_:-MS25MfQEedcd_zAILar:bhq9amo) ([raw file](https://github.com/pmem/rpma/blob/c978f2eac5f7b08d281725700f8ec075cce32c4c/tools/.gitignore#L4)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

> ```
> !report_bench.sh
> ```
what if we use an alternative name that does not need this exception i.e. rpma_fio_bench_report.sh?
</blockquote></details>

TBH I want to rename all scripts related to the report to start from common prefix e.g. `report_`.
I do not like `rpma_` prefix because this script is not related to `rpma_fio_bench.sh`. It uses it nothing more. But I am listening if you have other propositions. Why you do not like what I have proposed?
___
*[tools/report_bench.sh, line 17 at r1](https://reviewable.io/reviews/pmem/rpma/781#-MS1eQ18E_7Lq0VOVYfl:-MS268Yt1rq03Rxnj2J1:bdrovqq) ([raw file](https://github.com/pmem/rpma/blob/c978f2eac5f7b08d281725700f8ec075cce32c4c/tools/report_bench.sh#L17)):*
<details><summary><i>Previously, osalyk (Oksana Sałyk) wrote…</i></summary><blockquote>

What about REMOTE_DIRECT_WRITE_TO_PMEM?
</blockquote></details>

I have mentioned this variable only because from this level it is required. No changes to other variables. Maybe `REMOTE_NO_NOPASSWD` will appear here also at some point.
___
*[tools/report_bench.sh, line 31 at r1](https://reviewable.io/reviews/pmem/rpma/781#-MS1defs6a1MdPAd7loh:-MS26N3w50BCDUuHFgli:bd3i3x5) ([raw file](https://github.com/pmem/rpma/blob/c978f2eac5f7b08d281725700f8ec075cce32c4c/tools/report_bench.sh#L31)):*
<details><summary><i>Previously, osalyk (Oksana Sałyk) wrote…</i></summary><blockquote>

Isn't this variable redundant?  
</blockquote></details>

Both of these variables will be heavily used by all test cases I want to cover in this script.
At the same point, it is neat to have plain variables `DRAM` and `PMEM` and use them when they are needed. Don't you think?

**Review by osalyk — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/781#-:-MS2GZE2AdT-fl9vSqGv:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @janekmi)

File: tools/report_bench.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+#
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright 2021, Intel Corporation
+#
+
+#
+# report_bench.sh -- XXX (EXPERIMENTAL)
```
Comment (ldorau) [inline tools/report_bench.sh:8]: `XXX` should be replaced with a real description.

File: tools/report_bench.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	echo "Error: $1"
+	echo
+	echo "Usage: $0 <server_ip>"
+    echo
+	echo "export REMOTE_JOB_MEM_PATH=/path/to/mem (required)"
+	echo
+	exit 1
+}
+
+if [ "$#" -lt 1 ]; then
```
Comment (ldorau) [inline tools/report_bench.sh:22]: ` [ $# -lt 1 ]` - no "" - it is not a string

**Pull request merged**